### PR TITLE
4.x: Allow Unbuffered OutputStream for Http1ServerResponse (#9276)

### PR DESCRIPTION
**Pull request by VerKWer**
### Description
Currently, when building a new `WebServer` instance, one can specify `writeBufferSize(0)`. The server builds and starts normally, but any http1 request handler that tries to use `ServerResponse.outputStream()` will throw an exception. This is obviously not very user friendly, and we should either fail when trying to build/start the server instance or support unbuffered response `OutputStream`s.

I propose the latter approach, as it was the default before #6509 was merged. An obvious use case for unbuffered response writer is if the user does their own buffer management. For example, I'm currently moving an existing project to Helidon, which uses its own buffer pool to reduce GC pressure. Not allowing a buffer size of `0` would mean that there is no way to avoid the allocation of a new buffer for every request.

Solves issue #9276

### Documentation
I added a new abstract class `ResponseOutputStream` in `Http1ServerResponse`, which serves as a base for the two existing classes `BlockingOutputStream` and `ClosingBufferedOutputStream`. That way, we can choose which one to use after checking the write buffer size.

I also added the `WriteBufferTest` class to test both the default and zero write buffer case.

### CODE REVIEW

File: webserver/webserver/src/test/java/io/helidon/webserver/http1/WriteBufferTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class WriteBufferTest {
+
+    /**
+     * Test that a simple response can be sent using the {@link ServerResponse#outputStream()} using the default
+     * (non-zero) write buffer.
+     */
+    @Test
+    void defaultWriteBufferTest() throws Exception {
```
Comment (tomas-langer) [inline webserver/webserver/src/test/java/io/helidon/webserver/http1/WriteBufferTest.java:44]: Tests that require startup of a server should go to:
`webserver/tests/webserver`
where you can also use the `@ServerTest` annotation. See other tests in the module.
If you do not feel like changing the PR, let me know and I will take it over and do the rest.
Thanks for your contribution!

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: webserver/webserver/src/test/java/io/helidon/webserver/http1/WriteBufferTest.java (hunk 1 above)
Comment (VerKWer) [reply webserver/webserver/src/test/java/io/helidon/webserver/http1/WriteBufferTest.java:44]: Thanks for clarifying this. I'll gladly move the tests. How would I go about actually running the tests there? They don't seem to be part of the `webserver` module, and if I try using `webserver/tests` as a standalone project, compilation fails due to missing dependencies. I'm sure I'm missing something very obvious here.

Comment (VerKWer) [reply webserver/webserver/src/test/java/io/helidon/webserver/http1/WriteBufferTest.java]: Never mind, I forgot to activate the `tests` profile. I moved the relevant test to the correct module.

**Comment by tomas-langer** [comment]
Hello, thanks a lot for your contribution, but this was already fixed.
The fix is directly in constructor of `io.helidon.webserver.http1.Http1ServerResponse.ClosingBufferedOutputStream#ClosingBufferedOutputStream`
There is a new test: `io.helidon.webserver.tests.OutputStreamBufferSizeTest`

**Comment by romain-grecourt** [comment]
See #9313
